File: DijkstraAlgorithm.java
import java.util.*;

public class DijkstraAlgorithm {
    private int vertices;
    private List<List<Node>> adjacencyList;
    
    static class Node implements Comparable<Node> {
        int vertex, weight;
        
        Node(int vertex, int weight) {
            this.vertex = vertex;
            this.weight = weight;
        }
        
        public int compareTo(Node other) {
            return Integer.compare(this.weight, other.weight);
        }
    }
    
    public DijkstraAlgorithm(int vertices) {
        this.vertices = vertices;
        adjacencyList = new ArrayList<>();
        for (int i = 0; i < vertices; i++) {
            adjacencyList.add(new ArrayList<>());
        }
    }
    
    public void addEdge(int source, int destination, int weight) {
        adjacencyList.get(source).add(new Node(destination, weight));
        adjacencyList.get(destination).add(new Node(source, weight));
    }
    
    public void dijkstra(int source) {
        int[] distance = new int[vertices];
        Arrays.fill(distance, Integer.MAX_VALUE);
        distance[source] = 0;
        
        PriorityQueue<Node> pq = new PriorityQueue<>();
        pq.add(new Node(source, 0));
        
        while (!pq.isEmpty()) {
            Node current = pq.poll();
            int u = current.vertex;
            
            for (Node neighbor : adjacencyList.get(u)) {
                int v = neighbor.vertex;
                int weight = neighbor.weight;
                
                if (distance[u] + weight < distance[v]) {
                    distance[v] = distance[u] + weight;
                    pq.add(new Node(v, distance[v]));
                }
            }
        }
        
        // Print shortest distances
        for (int i = 0; i < vertices; i++) {
            System.out.println("Distance from " + source + " to " + i + ": " + distance[i]);
        }
    }
    
    public static void main(String[] args) {
        DijkstraAlgorithm graph = new DijkstraAlgorithm(6);
        graph.addEdge(0, 1, 4);
        graph.addEdge(0, 2, 3);
        graph.addEdge(1, 2, 1);
        graph.addEdge(1, 3, 2);
        graph.addEdge(2, 3, 4);
        graph.addEdge(3, 4, 2);
        graph.addEdge(4, 5, 6);
        
        graph.dijkstra(0);
    }
}

File: Python/graph/dijkstra_algorithm.py
import heapq
from collections import defaultdict

class DijkstraAlgorithm:
    def __init__(self):
        self.graph = defaultdict(list)
    
    def add_edge(self, source, destination, weight):
        self.graph[source].append((destination, weight))
        self.graph[destination].append((source, weight))
    
    def dijkstra(self, source, vertices):
        distance = [float('inf')] * vertices
        distance[source] = 0
        pq = [(0, source)]
        
        while pq:
            current_dist, u = heapq.heappop(pq)
            
            if current_dist > distance[u]:
                continue
            
            for v, weight in self.graph[u]:
                if distance[u] + weight < distance[v]:
                    distance[v] = distance[u] + weight
                    heapq.heappush(pq, (distance[v], v))
        
        # Print shortest distances
        for i in range(vertices):
            print(f"Distance from {source} to {i}: {distance[i]}")

if __name__ == "__main__":
    graph = DijkstraAlgorithm()
    graph.add_edge(0, 1, 4)
    graph.add_edge(0, 2, 3)
    graph.add_edge(1, 2, 1)
    graph.add_edge(1, 3, 2)
    graph.add_edge(2, 3, 4)
    graph.add_edge(3, 4, 2)
    graph.add_edge(4, 5, 6)
    
    graph.dijkstra(0, 6)